src/fifelse.c

```diff
@@ -16,17 +16,15 @@ SEXP fifelseR(SEXP l, SEXP a, SEXP b, SEXP na) {
   int nprotect = 0;
 
   if (ta != tb) {
-    if (ta == INTSXP && tb == REALSXP) {
-      SEXP tmp = PROTECT(coerceVector(a, REALSXP)); nprotect++;
-      a = tmp;
-      ta = REALSXP;
-    } else if (ta == REALSXP && tb == INTSXP) {
-      SEXP tmp = PROTECT(coerceVector(b, REALSXP)); nprotect++;
-      b = tmp;
-      tb = REALSXP;
-    } else {
-      error(_("'yes' is of type %s but 'no' is of type %s. Please make sure that both arguments have the same type."), type2char(ta), type2char(tb));
+    if (TYPEORDER(ta)<TYPEORDER(tb)) {
+      a = PROTECT(coerceAs(a, b, ScalarLogical(TRUE))); nprotect++;
+      ta = tb;
+    } else if (TYPEORDER(ta)>TYPEORDER(tb)) {
+      b = PROTECT(coerceAs(b, a, ScalarLogical(TRUE))); nprotect++;
+      tb = ta;
     }
+    if (TYPEOF(a)!=TYPEOF(b))
```

Comment:
i feel really not necessary. 

```
  SEXPTYPE ta = TYPEOF(a);
  SEXPTYPE tb = TYPEOF(b);
```

is 2 lines before

Reply:
it ensure that `coerceAs` produces matched types, won't cost much